src/modules/ekf2/EKF/control.cpp

```diff
@@ -227,7 +245,7 @@ void Ekf::controlExternalVisionFusion()
 		// determine if we should use the horizontal position observations
 		if (_control_status.flags.ev_pos) {
 			if (reset && _control_status_prev.flags.ev_pos) {
-				resetHorizontalPosition();
+				resetHorizontalPositionToVision();
```

Comment:
We need to be careful here, because the behavior changes when the filter is in GNSS + ev_pos ("delta pos/odometry" mode). When in that special case, we don't want to reset the position to vision, otherwise the GNSS data fusion will directly fail. Using ev_pos in odometry mode is more like a velocity aiding.

Maybe it's fine for now since it isn't used that much and we can fix is properly in your EV control refactoring PR.

Reply:
I added a check to only do the `resetHorizontalPositionToVision()` when `_fuse_hpos_as_odom` isn't set. Obviously not completely correct, but should be less wrong for now.